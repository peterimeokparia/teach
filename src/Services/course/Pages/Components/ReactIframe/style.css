.iframe {
    border-radius: 50px;
    -moz-border-radius: 50px;
}