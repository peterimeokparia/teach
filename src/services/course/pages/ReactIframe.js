import React, { useState, useEffect } from 'react';


const ReactIframe = ({  className, name, source, width, height, allow, scrolling, frameBorder}) => {

    return (
        <iframe
            name={name} 
            src={source}
            width={width}
            height={height}
            allow={allow}
            scrolling={scrolling}
            frameBorder={frameBorder}
            className={className}
            allowFullScreen
        >
       </iframe>
    );

}



export default ReactIframe;