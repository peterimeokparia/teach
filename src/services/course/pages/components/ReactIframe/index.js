import './style.css';

const ReactIframe = ({  
    className, 
    name, 
    source, 
    width, 
    height, 
    scrolling, 
    frameBorder }) => {
    return (
        <div className={"iframe"}>
            <iframe
                title={name}
                name={name} 
                src={source}
                width={width}
                height={height}
                allow="camera;microphone"
                scrolling={scrolling}
                frameBorder={frameBorder}
                className={className}
            >
            </iframe>
        </div>
       
    );
};

export default ReactIframe;