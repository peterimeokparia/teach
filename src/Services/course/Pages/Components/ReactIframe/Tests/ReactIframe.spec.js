import React from 'react';
import ReactIframe  from 'services/course/pages/components/ReactIframe';
import renderer from 'react-test-renderer';

describe('ReactIframe', () =>  {  

  beforeEach(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const tree = renderer.create(
        <ReactIframe /> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




