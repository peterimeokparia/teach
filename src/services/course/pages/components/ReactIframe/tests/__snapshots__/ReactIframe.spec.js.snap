// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactIframe renders as expected 1`] = `
<div
  className="iframe"
>
  <iframe
    allow="camera;microphone"
  />
</div>
`;
